### DISCUSSION TITLE
How do I add tempo markings?

### DISCUSSION THREAD

**Discussion by ego-lay-atman-bay** [0xfe/vexflow#1585] [Q&A] [answered]
I've tried adding tempo markings by using this
```javascript
let staveTempo = new StaveTempo(tempo.starting)
staveTempo.setTempo({
    bpm: tempo,
})
tempo.setContext(context).draw(stave)
```
But it doesn't seem to render. I checked in the dev tools, and there's no additional element for the tempo, which is why I'm here, I would like to know how to add tempo marking, because I can't find much about it.

edit: I just tried
```javascript
stave.setTempo(tempo)
```
which also didn't work

**Comment by ego-lay-atman-bay** [comment, marked as answer]
I just figured it out, I had to do this
```javascript
stave.setTempo({
    bpm: tempo
    duration: 'q',
})
```

**Reply by rvilarl** [reply]
Have a look at the tempo tests in stave_tests.ts
